app/mapping/playback/TrusteesExtractor.scala

```diff
@@ -0,0 +1,279 @@
+        Logger.error(s"[TrusteesExtractor] only address identification for lead trustee individual returned in DisplayTrustOrEstate api")
+        case object InvalidExtractorState extends RuntimeException
+        Failure(InvalidExtractorState)
+
+    } getOrElse {
+      Logger.error(s"[TrusteesExtractor] no identification for lead trustee individual returned in DisplayTrustOrEstate api")
+      case object InvalidExtractorState extends RuntimeException
+      Failure(InvalidExtractorState)
+    }
+  }
+
+
+  private def extractIndividualIdentification(individual: DisplayTrustTrusteeIndividualType, index: Int, answers: UserAnswers) = {
+    individual.identification map {
+
+      case DisplayTrustIdentificationType(_, Some(nino), None, None) =>
+        answers.set(TrusteeNinoYesNoPage(index), true)
+          .flatMap(_.set(TrusteesNinoPage(index), nino))
+
+      case DisplayTrustIdentificationType(_, None, Some(passport), None) =>
+        Logger.error(s"[TrusteesExtractor] only passport identification returned for trustee individual in DisplayTrustOrEstate api")
+        case object InvalidExtractorState extends RuntimeException
+        Failure(InvalidExtractorState)
+
+      case DisplayTrustIdentificationType(_, None, None, Some(address)) =>
+        answers.set(TrusteeNinoYesNoPage(index), false)
+          .flatMap(_.set(TrusteePassportIDCardYesNoPage(index), false))
+          .flatMap(answers => extractAddress(address.convert, index, answers))
+
+      case DisplayTrustIdentificationType(_, None, Some(passport), Some(address)) =>
+        answers.set(TrusteeNinoYesNoPage(index), false)
+          .flatMap(answers => extractAddress(address.convert, index, answers))
+          .flatMap(answers => extractPassportIdCard(passport, index, answers))
+
+    } getOrElse {
+      answers.set(TrusteeNinoYesNoPage(index), false)
+        .flatMap(_.set(TrusteeAddressYesNoPage(index), false))
+    }
+  }
+
+  private def extractLeadOrgIdentification(leadIndividual: DisplayTrustLeadTrusteeOrgType, index: Int, answers: UserAnswers) = {
+    leadIndividual.identification.map {
+
+      case DisplayTrustIdentificationOrgType(_, Some(utr), Some(address)) =>
+        answers.set(TrusteeUtrYesNoPage(index), true)
+          .flatMap(_.set(TrusteesUtrPage(index), utr))
+          .flatMap(answers => extractAddress(address.convert, index, answers))
+
+      case DisplayTrustIdentificationOrgType(_, None, Some(address)) =>
+        answers.set(TrusteeUtrYesNoPage(index), false)
+          .flatMap(answers => extractAddress(address.convert, index, answers))
+
+      case DisplayTrustIdentificationOrgType(_, Some(utr), None) =>
+        answers.set(TrusteeUtrYesNoPage(index), true)
+          .flatMap(_.set(TrusteesUtrPage(index), utr))
+
+    } getOrElse {
+      Logger.error(s"[TrusteesExtractor] no identification for lead trustee company returned in DisplayTrustOrEstate api")
+      case object InvalidExtractorState extends RuntimeException
```

Comment:
This case object is defined elsewhere, can we import and reuse that one?

Reply:
imported